rfcs/0000-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,86 @@
+# Buildpack.yml to Environment Variables
+
+## Proposal
+
+Migrate to using environment variables to do all buildpack configuration and
+get rid of `buildpack.yml`.
+
+## Motivation
+
+There are several reasons for making this switch.
+1. There is already an existing RFC that proposes moving away from
+   `buildpack.yml` as a configuration tool.
+1. Environment variables appears to be the standard for configuration in other
+   buildpack ecosystems such as Google Buildpacks and Heroku as well as the
+   Paketo Java buildpacks. Making this change will align the buildpack with the
+   rest of the buildpack ecosystem.
+1. There is native support to pass environment variables to the buildpack
+   either on a per run basis or by configuration that can be checked into
+   source control, in the form of `project.toml`.
+
+## Implementation
+
+The proposed environment variables for Go Build are as follow:
+
+#### BP_GO_BUILD_FLAGS
+
+```shell
+$BP_GO_BUILD_FLAGS='-buildmode=default,-tags=paketo,-ldflags="-X main.variable=some-value"'
```

Comment:
IMO, it would be more intuitive if these flags **aren't** comma delimited. Then they are copy-paste-able directly into the env var.

Reply:
Agree with @floragj. Also, I think using commas can get complicated. E.g. You can provide a comma-separated list as tags

Reply:
That seems reasonable to me and upon further investigation appears to be what [Go does for their environment variable](https://golang.org/cmd/go/#hdr-Environment_variables). I'll make this update right away.